Runtime/Tracking/Velocity/VelocityMultiplier.cs

```diff
@@ -21,13 +21,22 @@ public class VelocityMultiplier : VelocityTracker
         /// </summary>
         [Serialized]
         [field: DocumentedByXml]
-        public float VelocityMultiplierFactor { get; set; } = 1f;
+        public Vector3 VelocityMultiplierFactor { get; set; } = Vector3.one;
         /// <summary>
         /// The amount to multiply the source angular velocity by.
         /// </summary>
         [Serialized]
         [field: DocumentedByXml]
-        public float AngularVelocityMultiplierFactor { get; set; } = 1f;
+        public Vector3 AngularVelocityMultiplierFactor { get; set; } = Vector3.one;
+
+        /// <summary>
+        /// The container for muliplied velocity.

```

Comment:
Why is this caching of the results necessary? Can we remove these two `protected` fields?

Reply:
Just so it's not doing a `new Vector3` in place

because the source is a return value so you can't do

`source.GetVelocity().Scale(vec)`

because source.GetVelocity is a calculation returned from another method

Reply:
These are value types, the copying is happening anyway. Use a local variable in the methods instead, e.g. `Vector3 multipliedVelocity = Source.GetVelocity();`.